### PR TITLE
Add e2e-serial jobs to all network-edge component repos

### PR DESCRIPTION
**Pull request by sgreene570**
Add e2e-aws-serial jobs to:
 * cluster-dns-operator
 * router

Add e2e-gcp-serial jobs to:
 * cluster-ingress-operator
 * coredns

 Note that these jobs are required for merging to these repos
 as per @smarterclayton's request.

### CODE REVIEW

File: ci-operator/config/openshift/coredns/openshift-coredns-master.yaml [hunk 1]
```diff
@@ -75,6 +75,10 @@ tests:
   steps:
     cluster_profile: aws
     workflow: openshift-upgrade-aws
+- as: e2e-aws-serial
```
Comment (smarterclayton) [inline ci-operator/config/openshift/coredns/openshift-coredns-master.yaml:78]: make these GCP

File: ci-operator/config/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master.yaml [hunk 2]
```diff
@@ -103,6 +103,10 @@ tests:
   steps:
     cluster_profile: aws
     workflow: openshift-upgrade-aws
+- as: e2e-aws-serial
```
Comment (smarterclayton) [inline ci-operator/config/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master.yaml:106]: make this gcp

Comment (sgreene570) [reply ci-operator/config/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master.yaml]: done.

File: ci-operator/config/openshift/coredns/openshift-coredns-master.yaml (hunk 1 above)
Comment (sgreene570) [reply ci-operator/config/openshift/coredns/openshift-coredns-master.yaml]: done.

**Comment by sgreene570** [comment]
messed up the aws-> gcp switch
/hold

**Comment by sgreene570** [comment]
fixed
/hold cancel

File: ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master-presubmits.yaml [hunk 3]
```diff
@@ -402,6 +402,74 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-gcp-operator,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
+    branches:
+    - master
+    cluster: build02
+    context: ci/prow/e2e-gcp-serial
```
Comment (Miciah) [inline ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master-presubmits.yaml:410]: Elsewhere I see `context: ci/prow/k8s-e2e-gcp-serial` (and `mountPath: /usr/local/k8s-e2e-gcp-serial-cluster-profile`, `--target=k8s-e2e-gcp-serial`, and so on); should this have `k8s` in the name?

File: ci-operator/jobs/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master-presubmits.yaml [hunk 4]
```diff
@@ -132,6 +132,72 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-aws-operator,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
+    branches:
+    - master
+    cluster: build01
```
Comment (Miciah) [inline ci-operator/jobs/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master-presubmits.yaml:139]: Should this be build02?

Comment (sgreene570) [reply ci-operator/jobs/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master-presubmits.yaml:139]: If I change this `build02`, running `make jobs` reverts the change.

File: ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master-presubmits.yaml (hunk 3 above)
Comment (sgreene570) [reply ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master-presubmits.yaml:410]: Hm, what other repos where you looking at that had `k8s-e2e-gcp-serial`? Kube-component repos?

None of the definitions in our repos use the `ci/prow/k8s-*` pattern, so I think this is fine.

Comment (Miciah) [reply ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master-presubmits.yaml:410]: When I ran `git -C ~/src/github.com/openshift/release grep -e 'context: ci/prow/e2e-gcp-serial'`, the only results I saw were for cluster-ingress-operator and coredns (I had your PR checked out locally).  When I ran `git -C ~/src/github.com/openshift/release grep -e 'context: ci/prow/k8s-e2e-gcp-serial'`, I saw results for cluster-kube-apiserver-operator and kubernetes.  Maybe it's a kube-component thing and not an e2e-serial thing.

Comment (Miciah) [reply ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master-presubmits.yaml:410]: When I ran `git -C ~/src/github.com/openshift/release grep -e 'context: ci/prow/e2e-gcp-serial'`, the only results I saw were for cluster-ingress-operator and coredns (I had your PR checked out locally).  When I ran `git -C ~/src/github.com/openshift/release grep -e 'context: ci/prow/k8s-e2e-gcp-serial'`, I saw results for cluster-kube-apiserver-operator and kubernetes.  Maybe it's a kube-component thing and not an e2e-serial thing.

Comment (smarterclayton) [reply ci-operator/jobs/openshift/cluster-ingress-operator/openshift-cluster-ingress-operator-master-presubmits.yaml:410]: No it should not have k8s in the name.  That's only for openshift/kubernetes

**Pull request merged**